components/external-services-mock/cmd/main.go

```diff
@@ -212,34 +210,50 @@ func initDefaultServer(cfg config, key *rsa.PrivateKey) *http.Server {
 	}
 }
 
+func initDefaultCertServer(cfg config, key *rsa.PrivateKey) *http.Server {
+	router := mux.NewRouter()
+
+	// Oauth server handlers
+	tokenHandlerWithKey := oauth.NewHandlerWithSigningKey(cfg.ClientSecret, cfg.ClientID, cfg.TenantHeader, key)
+	// TODO The mtls_token_provider sends client id and scopes in url.values form. When the change for fetching xsuaa token
+	// with certificate is merged GenerateWithCredentialsFromReqBody should be used for testing the flows that include fetching
+	// xsuaa token with certificate. APP_SELF_REGISTER_OAUTH_TOKEN_PATH for local env should be adapted.
+	router.HandleFunc("/oauth/token", tokenHandlerWithKey.GenerateWithCredentialsFromReqBody).Methods(http.MethodPost)
+
+	tokenHandler := oauth.NewHandler(cfg.ClientSecret, cfg.ClientID)
+	router.HandleFunc("/cert/token", tokenHandler.GenerateWithoutCredentials).Methods(http.MethodPost)
```

Comment:
Why we need both token handlers with cert?

Reply:
We may unify them, they were introduced from another changes/PRs and I transfer them in the "default cert server" but I can check what need to be change to remove one of them